hedera-mirror-web3/src/main/java/com/hedera/mirror/web3/evm/token/TokenAccessorImpl.java

```diff
@@ -0,0 +1,405 @@
+                    };
+        }
+        return new EvmKey();
+    }
+
+    @Override
+    public String nameOf(final Address token) {
+        final var tokenId = entityIdFromEvmAddress(token);
+        return tokenRepository.findName(tokenId).orElse("");
+    }
+
+    @Override
+    public String symbolOf(final Address token) {
+        final var tokenId = entityIdFromEvmAddress(token);
+        return tokenRepository.findSymbol(tokenId).orElse("");
+    }
+
+    @Override
+    public long totalSupplyOf(final Address token) {
+        final var tokenId = entityIdFromEvmAddress(token);
+        return tokenRepository.findTotalSupply(tokenId).orElse(0L);
+    }
+
+    @Override
+    public int decimalsOf(final Address token) {
+        final var tokenId = entityIdFromEvmAddress(token);
+        return tokenRepository.findDecimals(tokenId).orElse(0);
+    }
+
+    @Override
+    public long balanceOf(Address account, Address token) {
+        final var tokenId = entityIdFromEvmAddress(token);
+        final var accountId = entityIdFromAccountAddress(account);
+        return tokenBalanceRepository.findBalance(tokenId, accountId).orElse(0L);
+    }
+
+    @Override
+    public long staticAllowanceOf(final Address owner, final Address spender, final Address token) {
+        final var tokenId = entityIdFromEvmAddress(token);
+        final var ownerId = entityIdFromAccountAddress(owner);
+        final var spenderId = entityIdFromAccountAddress(spender);
+        return tokenAllowanceRepository.findAllowance(tokenId, ownerId, spenderId).orElse(0L);
+    }
+
+    @Override
+    public Address staticApprovedSpenderOf(final Address nft, long serialNo) {
+        final var tokenId = entityIdFromEvmAddress(nft);
+        final var spenderNum = nftRepository.findSpender(tokenId, serialNo);
+        if (spenderNum.isEmpty()) {
+            return Address.ZERO;
+        }
+        final var spenderEntity = EntityIdEndec.decode(spenderNum.get(), ACCOUNT);
+        return Address.wrap(Bytes.wrap(toEvmAddress(spenderEntity)));
+    }
+
+    @Override
+    public boolean staticIsOperator(final Address owner, final Address operator,
+                                    final Address token) {
+        final var tokenId = entityIdFromEvmAddress(token);
+        final var ownerId = entityIdFromAccountAddress(owner);
+        final var spenderId = entityIdFromAccountAddress(operator);
+        final var isSpenderAnOperator = nftAllowanceRepository.isSpenderAnOperator(tokenId, ownerId, spenderId);
+
+        return isSpenderAnOperator != null && isSpenderAnOperator;
+    }
+
+    @Override
+    public Address ownerOf(final Address nft, long serialNo) {
+        final var tokenId = entityIdFromEvmAddress(nft);
+        final var ownerNum = nftRepository.findOwner(tokenId, serialNo);
+        if (ownerNum.isEmpty()) {
+            return Address.ZERO;
+        }
+        final var ownerEntity = EntityIdEndec.decode(ownerNum.get(), ACCOUNT);
```

Comment:
Don't use EntityIdEndec directly.
```suggestion
        final var ownerEntity = EntityId.of(ownerNum.get(), ACCOUNT);
```